### ISSUE TITLE
@Retry has time unit attributes with names inconsistent with the main attribute name

### ISSUE THREAD

**Issue by Ladicek** [eclipse/microprofile-fault-tolerance#524]
We mostly use a convention that a time unit attribute for some attribute `foo` is called `fooUnit` (for example, `delay` and `delayUnit`). Perhaps with an exception that if the attribute is called `value`, then the time unit attribute is just `unit`.

However, some `@Retry` attributes don't stick to this convention:

- `maxDuration` and `durationUnit`
- `jitter` and `jitterDelayUnit`

Especially `durationUnit` might look like perhaps it covers all duration values, and not just `maxDuration`. This was pointed out by @yoshioterada here: https://github.com/smallrye/smallrye-fault-tolerance/issues/206

I propose we rename `durationUnit` to `maxDurationUnit`, and `jitterDelayUnit` to `jitterUnit`. We can do it in 2 ways:

1. Hard breaking change. Just do the rename.
2. With a deprecation period. For some period of time, both variants would exist, the old one being `@Deprecated`. If both were used, the new one would have a priority. This should be relatively easy to implement, as there's no collision.

**Comment by Emily-Jiang** [comment]
Yes, we discussed this unit renaming before but dismissed the suggestion of renaming the parameters as the names are straightforwad and not worth the effor of breaking backward compatibility. However, I think the 2nd option sounds like a good compromise.

**Comment by Ladicek** [comment]
I would agree that "the names are straightforwad", but I received a bug report, so clearly there is _some_ problem :-)

I'll try implementing the 2nd option and see how it looks like.
